File: FitHub.Web/Modeles/Identity/RegistrationModel.cs
using System.ComponentModel.DataAnnotations;

namespace FitHub.Web.Modeles.Identity
{
    public class RegistrationModel
    {
        [Required(ErrorMessage = "Email обязателен")]
        [EmailAddress(ErrorMessage = "Введите действительный Email адрес")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Пароль обязателен")]
        [StringLength(100, ErrorMessage = "Пароль должен содержать не менее {2} символов.", MinimumLength = 6)]
        [DataType(DataType.Password)]
        public string Password { get; set; }

        [Required(ErrorMessage = "Подтвердите пароль")]
        [DataType(DataType.Password)]
        [Compare("Password", ErrorMessage = "Пароли не совпадают.")]
        public string ConfirmPassword { get; set; }

        [Required]
        public string Firstname { get; set; }

        [Required]
        public string Lastname { get; set; }

        public DateOnly Birthdate { get; set; }

        public string Location { get; set; }

        public List<int> Usersports { get; set; }
    }
}

File: app/src/main/java/com/example/fithub/models/RegistrationModel.kt
package com.example.fithub.models

data class RegistrationModel(
    //@field:NotEmpty(message = "Email обязателен")
    //@field:Email(message = "Введите действительный Email адрес")
    val email: String,

    //@field:NotEmpty(message = "Пароль обязателен")
    //@field:Size(min = 6, message = "Пароль должен содержать не менее 6 символов")
    val password: String,

    //@field:NotEmpty(message = "Подтвердите пароль")
    val confirmPassword: String,

    //@field:NotEmpty(message = "Имя обязательно")
    val firstname: String,

    //@field:NotEmpty(message = "Фамилия обязательна")
    val lastname: String,

    val birthdate: String? = null,

    val location: String? = null
)
